src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -255,13 +337,25 @@ public static bool UpdateInstallerNodes(InstallerManifest installerManifest, IEn
             }
         }
 
-        private static bool ParsePackageAndGenerateInstallerNodes(string path, string url, List<Installer> installers, Manifests manifests)
+        private static bool ParsePackageAndGenerateInstallerNodes(
+            string path,
+            string url,
+            List<Installer> installers,
+            Manifests manifests)
         {
             var installer = new Installer();
             installer.InstallerUrl = url;
             installer.InstallerSha256 = GetFileHash(path);
             installer.Architecture = GetMachineType(path)?.ToString().ToEnumOrDefault<InstallerArchitecture>() ?? InstallerArchitecture.Neutral;
 
+            var archGuess = GetArchFromUrl(installer.InstallerUrl);
```

Comment:
At this point, we've only tried to parse the arch from the exe, not an MSI. This needs to be after line 361

Reply:
I missed this. moved logic to after line 361

Reply:
Only for exe/msi. MSIX we shouldn't override

Reply:
made changes so that architecture override only occurs for exe/msi and not MSIX